modules/vectra.py

```diff
@@ -85,19 +85,20 @@ def __init__(self, url=None, token=None, user=None, password=None, verify=False)
         *Either token or user are required
         """
         self.url = url
-        self.version = 2 if token else 1
         self.verify = verify
 
         url = VectraClient._remove_trailing_slashes(url)
 
         if token:
-            self.url = f'{url}/api/v2'
+            self.version = 2
```

Comment:
Suggestion.

If you make the version or VERSION2 as a class variable, then the child classes only need to override the  class variable in the class definition.  This avoids the need to override the init() method on the child classes.

```
class BaseClass():
   VERSION2 = 2
   VERSION1 = 1
    
    def __init__(self, url=None, token=None, user=None, password=None, verify=False):
        ...
        if token and self.VERSION2 is not None:
            self.version = self.VERSION2
            ....
        elif user and password  and self.VERSION1 is not None:
            self.version = self.VERSION1
            ....
         else:
             raise()
             
             
             
 class ChildClass(BaseClass):
     VERSION2 = 2.4
     VERSION1 = None
```